libnetwork/drivers/bridge/setup_ip_tables_linux_test.go

```diff
@@ -138,4 +151,37 @@ func assertBridgeConfig(config *networkConfiguration, br *bridgeInterface, d *dr
 	if err != nil {
 		t.Fatalf("%v", err)
 	}
+	if d.config.EnableIP6Tables {
+		if err := nw.setupIP6Tables(config, br); err != nil {
+			t.Fatalf("%v", err)
+		}
+	}
+}
+
+// Regression test for https://github.com/moby/moby/issues/46445
+func TestSetupIP6TablesWithHostIP(t *testing.T) {
+	defer netnsutils.SetupTestOSContext(t)()
+	d := newDriver()
+	dc := &configuration{
+		EnableIPTables:  true,
+		EnableIP6Tables: true,
+	}
+	if err := d.configure(map[string]interface{}{netlabel.GenericData: dc}); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
nit: perhaps inline `dc` as it's only used in a single place;

```suggestion
	if err := d.configure(map[string]interface{}{netlabel.GenericData: &configuration{
		EnableIPTables:  true,
		EnableIP6Tables: true,
	}}); err != nil {
		t.Fatal(err)
	}
```